doc/src/read_data.rst

```diff
@@ -364,24 +365,41 @@ periodic remapping will be performed using simulation box bounds that
 are the union of the existing box and the box boundaries in the new
 data file.
 
+If the system is non-periodic (in a dimension), then an image flag has
+no meaning (in that dimension) and can lead to unexpected results for
+computations that internally use unwrapped coordinates (like the center of mass).
+Thus those will be be reset to zero on reading and LAMMPS will print a
+warning, if that happens.  If those atoms with non-zero image flags are
+involved in bonded interactions, this simple reset can lead to errors.
```

Comment:
This implies that not resetting the image flags to 0 would
not cause problems or errors.  But that's not true.  Those
bonds would be a box-size in length.  So I'd rather say you've
got a problem either way if you import PBC straddling bonds into a new non-periodic box.

Reply:
@sjplimp. There is already text in the following lines explaining the situation.
On top of that, I've been looking into ways on how to do the folding back of molecules into the principal cell with LAMMPS scripting and I got it about halfway done. The "data.spce" file in "examples/USER/tally" is a case where you have bonds and non-zero image flags due to it being extracted from a long equilibration. Just setting the image flags to 0 will break the structure. The following input code "fixes" this geometry by wrapping molecules back on 3 of the 6 sides of the box. Just visualize the structure (with the topology defined bonds) before and after.
```
atom_style full
boundary p p p
pair_style zero 4.0
bond_style zero
angle_style zero

read_data data.spce

pair_coeff * *
bond_coeff * 1.0
angle_coeff * 109.0
                          
compute imgx all property/atom ix
compute imgy all property/atom iy
compute imgz all property/atom iz
compute molchunk all chunk/atom molecule 

compute molimg all reduce/chunk molchunk min c_imgx c_imgy c_imgz
compute redimgx all chunk/spread/atom molchunk c_molimg[1]
compute redimgy all chunk/spread/atom molchunk c_molimg[2]
compute redimgz all chunk/spread/atom molchunk c_molimg[3]

variable molimgx atom c_imgx-c_redimgx
variable molimgy atom c_imgy-c_redimgy
variable molimgz atom c_imgz-c_redimgz

fix 1 all ave/atom 1 1 1 v_molimgx v_molimgy v_molimgz
run 0 pre no post no

set atom * image  v_molimgx v_molimgy v_molimgz
write_data data.lower
```
I didn't get it to work properly on the other 3 sides because of problems with multiple "run 0" commands and having to reset/retrigger computes and using a fix to "consume" the computes to avoid the errors when running computes between runs.

But I think this can be handled by adding a command `reset_image_flags` that can bypass the issues and augment problems of moving the data with explicit MPI calls. If need be, this would require to run `reset_mol_ids` first to have molecule IDs match the topological grouping. With such a command available, the documentation would be more straightforward and just recommend using that or external processing of unwrapped coordinates.

Reply:
Just to be clear about the procedure. When you have a molecular structure the sequence would be:
- use read_data with periodic boundaries
- use reset_image_flags to reset image flags to as small as possible values (typically -1, 0, or 1)
- use change_box to expand the box in z (or x, or y) so that image flags in that direction become 0
- use change_box to switch to non-periodic in z
The key point is that image flags must be 0 before switching periodicity off in that direction.

Reply:
> see my comment in the read_data doc page

@sjplimp I have reformulated the paragraphs about resetting image flags on reading a data file and added a reference to the `change_box` command. This should be ready to go now.